contracts/beyondFramework/dca/libraries/DCAHistoryLib.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+library DCAHistoryLib {
+    struct HistoricalGauge {
+        uint256 amountSpent;
+        uint256 amountExchanged;
+    }
+
+    struct DCAHistory {
+        HistoricalGauge[] gauges;
+        uint256 current;
+    }
+
+    function addHistoricalGauge(
+        DCAHistory storage history,
+        uint256 amountSpent,
+        uint256 amountExchanged
+    ) internal {
+        history.gauges.push(HistoricalGauge(amountSpent, amountExchanged));
+        history.current++;
+    }
+
+    function increaseGaugeAt(
+        DCAHistory storage history,
+        uint256 rewards,
+        uint256 index
+    ) internal {
+        history.gauges[index].amountExchanged += rewards;
+    }
+
+    function decreaseGaugeByIndex(
+        DCAHistory storage history,
+        uint256 index,
+        uint256 amountSpent,
+        uint256 amountExchanged
+    ) internal {
+        history.gauges[index].amountSpent -= amountSpent;
+        history.gauges[index].amountExchanged -= amountExchanged;
+    }
+
+    function currentHistoricalIndex(DCAHistory storage history)
```

Comment:
Likewise, HistoricalIndex and Index are mixed. Of course, choice is up to you :)